Psychtoolbox/PsychHardware/EyelinkToolbox/EyelinkDemos/SR-ResearchDemos/SimpleVideo/EyeLink_SimpleVideo.m

```diff
@@ -0,0 +1,396 @@
+    
+    
+    % Configure animated calibration target path and properties
+    el.calTargetType = 'video';
+    calMovieName = ('calibVid.mov');
+    
+    el.calAnimationTargetFilename = [pwd '/' calMovieName];
+    el.calAnimationResetOnTargetMove = true; % false by default, set to true to rewind/replay video from start every time target moves
+    el.calAnimationAudioVolume = 0.4; % default volume is 1.0, but too loud on some systems. Setting volume lower to 0.4 (minimum is 0.0)
+    
+    % You must call this function to apply the changes made to the el structure above 
+    EyelinkUpdateDefaults(el);
+    
+    % Set display coordinates for EyeLink data by entering left, top, right and bottom coordinates in screen pixels
+    Eyelink('Command','screen_pixel_coords = %ld %ld %ld %ld', 0, 0, width-1, height-1);
+    % Write DISPLAY_COORDS message to EDF file: sets display coordinates in DataViewer
+    % See DataViewer manual section: Protocol for EyeLink Data to Viewer Integration > Pre-trial Message Commands
+    Eyelink('Message', 'DISPLAY_COORDS %ld %ld %ld %ld', 0, 0, width-1, height-1);    
+    % Set number of calibration/validation dots and spread: horizontal-only(H) or horizontal-vertical(HV) as H3, HV3, HV5, HV9 or HV13
+    Eyelink('Command', 'calibration_type = HV5'); % horizontal-vertical 5-points
+    % Allow a supported EyeLink Host PC button box to accept calibration or drift-check/correction targets via button 5
+    Eyelink('Command', 'button_function 5 "accept_target_fixation"');
+    % Hide mouse cursor
+    HideCursor(screenNumber);
+    % Start listening for keyboard input. Suppress keypresses to Matlab windows.
+    ListenChar(-1);
+    % Clear Host PC display from any previus drawing
+    Eyelink('Command', 'clear_screen 0');
+    % Put EyeLink Host PC in Camera Setup mode for participant setup/calibration
+    EyelinkDoTrackerSetup(el);
+    
+    
+    %% STEP 5: TRIAL LOOP.
+    
+    driftVidList = {'dotsGrey.mov' 'wheelGrey.mov'};% Provide drift-check video file list for 2 trials
+    vidList = {'expected.mov' 'disappear.mov'};% Provide trial video file list for 2 trials
+    
+    spaceBar = KbName('space');% Identify keyboard key code for space bar to end each trial later on    
+    for i = 1:length(vidList)
+           
+        % Change animated calibration target path for drift-check/correction
+        calMovieName = char(driftVidList(i));
+        el.calAnimationTargetFilename = [pwd '/' calMovieName];       
+        % You must call this function to apply the changes made to the el structure above
+        EyelinkUpdateDefaults(el);
+        
+        % Open movie file:
+        movieName = char(vidList(i));
+        moviePath = [ pwd '/' movieName ];
+        [movie, ~, ~, Movx, Movy] = Screen('OpenMovie', window, moviePath, [], [], spcf1); % spcf1 required to disable audio on macOS Catalina and avoid playback freezing issues

```

Comment:
Wrt. to the lag between showing 1st movie frame and successive frames as mentioned above:

In this script one can work around this and essentially remove the lag, by placing the 'OpenMovie' command immediately before the 'PlayMovie' command below, as due to the underlying bug the duration of the "freeze" is the duration in time between calling OpenMovie and PlayMovie. Placing them back to back reduces the freeze to at most a few imperceptible millsecond(s).

Reply:
resolved in 0abf0cbe36691edea3a6d49a87b79149ad7e6f94 moved OpenMovie to after long/blocking call to EyelinkDoDriftCorrection to reduce this interval as much as possible. However, MovX/MovY returned by OpenMovie are required for pre-trial commands that draw to the Host PC, which must be done/processed by Host PC prior to StartRecording